staging/src/k8s.io/apiserver/pkg/endpoints/filters/storageversion.go

```diff
@@ -69,6 +73,31 @@ func WithStorageVersionPrecondition(handler http.Handler, svm storageversion.Man
 			handler.ServeHTTP(w, req)
 			return
 		}
+		// The system namespace is required for apiserver-identity lease to exist. Allow the apiserver
+		// itself to create namespaces.
+		// NOTE: with this exception, if the bootstrap client writes namespaces with a new version,
```

Comment:
I'm nervous that there may be more dependencies like this, this feature is behind a gate, right?

Reply:
two alpha feature gates are required: StorageVersionAPI and APIServerIdentity

I have tested locally, allowing `namespaces` and `leases` writes stops this feature from deadlock itself.

we can get signal of the filter breaking any bootstrap process from the alpha e2e test
/test pull-kubernetes-e2e-gce-alpha-features